### PR TITLE
cached decorator takes query_string parameter

### PR DESCRIPTION
**Pull request by lily-mayfield**
Provide a keyword `query_string` to `cached` decorator in order
to create the same cache key for different query string requests,
so long as they have the same key/value (order does not matter).

Create tests for aforementioned.

### CODE REVIEW

**Comment by lily-mayfield** [comment]
Looks like there's an error for Python 3.5 (though this fix works for Python 3.5.2 on the https://github.com/opensyllabus/osp-api project and is tested).

These errors come up...
```
FAIL: test_10a_arg_kwarg_memoize_var_keyword (test_cache.CacheFilesystemTestCase)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/sh4nks/flask-caching/test_cache.py", line 363, in test_10a_arg_kwarg_memoize_var_keyword

    assert f(1, 2, d=5, e=8) == f(1, 2, e=8, d=5)

AssertionError

======================================================================

FAIL: test_10a_arg_kwarg_memoize_var_keyword (test_cache.CacheRedisTestCase)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/sh4nks/flask-caching/test_cache.py", line 363, in test_10a_arg_kwarg_memoize_var_keyword

    assert f(1, 2, d=5, e=8) == f(1, 2, e=8, d=5)

AssertionError

======================================================================

FAIL: test_10a_arg_kwarg_memoize_var_keyword (test_cache.CacheTestCase)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/sh4nks/flask-caching/test_cache.py", line 363, in test_10a_arg_kwarg_memoize_var_keyword

    assert f(1, 2, d=5, e=8) == f(1, 2, e=8, d=5)

AssertionError
```

**Comment by sh4nks** [comment]
Thanks, I appreciate it!
btw, I am not sure why the tests are failing (see #27)
Right now I am in the process of rewriting the tests to use pytest and to be more modular - after that I'll try to refactor the code a bit because it's kind of a mess at the moment.

**Comment by lily-mayfield** [comment]
My pleasure @sh4nks! Thanks for being so communicative and expeditious!

File: test_cache.py [hunk 1]
```diff
@@ -785,6 +785,65 @@ def view_pie(flavor):
...
+        cache/cache key:
+
+          * GET /v1/works?mock=true&offset=20&limit=15
+          * GET /v1/works?limit=15&mock=true&offset=20
+
+        Caching functionality is verified by time from response included
+        with any cache-able resource response.  The time in the response
+        can verify that two requests with the same query string
+        parameters/values, though differently ordered, produce responses
+        with the same `cache_timestamp`.
```
Comment (lily-mayfield) [inline test_cache.py]: The same time, not `cache_timestmap`.

File: test_cache.py [hunk 2]
```diff
@@ -785,6 +785,65 @@ def view_pie(flavor):
...
+        any/all query string parameters.
+
+        For example, these two requests should yield the same
+        cache/cache key:
+
+          * GET /v1/works?mock=true&offset=20&limit=15
+          * GET /v1/works?limit=15&mock=true&offset=20
+
+        Caching functionality is verified by time from response included
+        with any cache-able resource response.  The time in the response
```
Comment (lily-mayfield) [inline test_cache.py]: verified by a `@cached` route `/works` which produces a time in its response.

**Comment by sh4nks** [comment]
I have updated the docstring - I think it's fine now.
See 2cd1722d50f92d847737609425810773eaa4ff67.

**Comment by sh4nks** [comment]
btw, I have pushed version 1.3.0 to pypi.

**Comment by lily-mayfield** [comment]
@sh4nks That's wonderful! I'll use that on `osp-api`! Thanks!

**Pull request merged**